internal/util/test/repo.go

```diff
@@ -0,0 +1,17 @@
+package test
+
+import (
+	"context"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func getRepoRoot(ctx context.Context) (string, error) {
+	out, err := exec.CommandContext(ctx, "git", "rev-parse", "--show-toplevel").Output()
+	if err != nil {
+		return "", err
+	}
+	dir := strings.TrimSpace(filepath.Clean(string(out)))
+	return dir, nil
+}
```

Comment:
An alternative without dependency on git and context:
```suggestion
// RepoRoot returns the absolute path to the root of the repository.
func RepoRoot() string {
	_, b, _, _ := runtime.Caller(0)
	d := filepath.Dir(path.Join(path.Dir(b), "../..")) // Number of ../ depends on the path of this file.
	root, _ := filepath.Abs(d)
	return root
}
```
I think there's no point in returning errors from it as it is quite "static" in the context of the repository.

BTW this could be useful for adding helper functions reading test dependencies versions from `.github/test_dependencies.yaml`.

Reply:
> I think there's no point in returning errors from it as it is quite "static" in the context of the repository.

git call can always fail so it's better to catch errors early.

Reply:
> git call can always fail so it's better to catch errors early.

I was referring to the alternative version I posted that doesn't rely on git - `filepath.Abs` technically returns an error, but it's unlikely to break in the context of the repository.

Reply:
Done, change to use `filepath` as suggested.